delphi/program_analysis/ProgramAnalysisGraph.py

```diff
@@ -8,69 +8,88 @@
 from inspect import signature
 
 
+def add_variable_node(G, n):
+    """ Add a variable node to the CAG. """
+    name = n.attr["cag_label"]
+    G.add_node(
+        name,
+        value=None,
+        pred_fns=[],
+        agraph_name=n,
+        index=n.attr["index"],
+        node_type=n.attr["node_type"],
+        start=n.attr["start"],
+        end=n.attr["end"],
+        index_var=n.attr["index_var"],
+        visited=False,
+    )
+
+    # If the node is a loop index, set special initialization
+    # and update functions.
+    if n.attr["is_index"] == "True":
+        G.nodes[name]["is_index"] = True
+        G.nodes[name]["value"] = int(n.attr["start"])
+        G.nodes[name]["visited"] = True
+        G.nodes[name]["update_fn"] = (
+            lambda **kwargs: int(kwargs.pop(list(kwargs.keys())[0])) + 1
+        )
+        G.add_edge(name, name)
+
+
+def add_action_node(A: AGraph, G: nx.DiGraph, λs, n):
```

Comment:
Not so wild about having unicode characters in our code. This seems like a large decrease in writability to me as it takes time to learn letter locations for different keymaps. It takes much less time for me to type `lambdas` when trying to reference a variable name than it does for me to google how to type the character lambda on a mac (and I am unwilling to just copy-pasta a character).

Also using lambda here gives me the first-impression that this variable is related to the mathematical usage of lambda; however, I believe this particular usage is closer to the colloquial CS usage.

Please see this [StackExchange post](https://softwareengineering.stackexchange.com/questions/16010/is-it-bad-to-use-unicode-characters-in-variable-names) for an extended discussion on this matter.

Reply:
Fair enough (although lambda is descended from the lambda calculus, which was invented by a mathematician :) )

I have no problem removing the unicode character though.